src/features/notifications/components.tsx

```diff
@@ -1,11 +1,12 @@
 import { AppNotification } from 'common/models/notifications';
 import { FC } from 'react';
 import Moment from 'react-moment';
-import { Link } from 'react-router-dom';
+import { NotificationLink } from './components/NotificationLink';
 
 export const AgentCreatedNotification: FC<{
   notification: AppNotification;
-}> = ({ notification }) => {
+  handleMarkRead?: (notificationId: number) => void;
```

Comment:
`renderNotification` is always going to be passed `handleMarkRead` so this shouldn't be optional, we're creating an interface here, so making this optional just makes the developers life harder when implementing notifications.

Reply:
That's not correct. It isn't passed handleMarkRead for the ReadNotifications component.

Reply:
In that case, `handleMarkRead` could take a `Notification` instead of a `id` and do the right thing based on whether the notification is read or not. The notification component itself shouldn't really have to care whether it's been read or not.

Reply:
Ah, okay. I see what you mean. Okay. I'll go ahead and refactor this.